File: Capstone Project/Backend side/AssessmentPortal/src/main/java/com/NTeq/AssessmentPortal/Services/CategoryService.java
package com.NTeq.AssessmentPortal.Services;

import java.util.List;

import com.NTeq.AssessmentPortal.Dto.CategoryDto;
import com.NTeq.AssessmentPortal.Response.SuccessResponse;
/**
 * Service interface for managing category-related operations.
 */
public interface CategoryService {
    /**
     * Adds a new category to the system.
     * @param cgDto The DTO containing category details.
     * @return A message indicating the result of the operation.
     */
     SuccessResponse addCategory(CategoryDto cgDto);
    /**
     * Retrieves a list of all categories.
     * @return A list of category DTOs.
     */
    List<CategoryDto> getAllCategory();
    /**
     * Retrieves a category by its unique identifier.
     * @param categoryId The ID of the category to retrieve.
     * @return The DTO representing the retrieved category.
     */
    CategoryDto getCategoryById(long categoryId);
    /**
     * Updates a category's information.
     * @param categoryId The ID of the category to update.
     * @param cgDto      The DTO containing updated category details.
     * @return A message indicating the result of the operation.
     */
    SuccessResponse updateCategory(Long categoryId, CategoryDto cgDto);
    /**
     * Deletes a category by its unique identifier.
     * @param categoryId The ID of the category to delete.
     * @return A message indicating the delete operation.
     */
    SuccessResponse deleteCategory(long categoryId);
}

File: Capstone Project/assessment-portal-frontend/src/Services/categoryService.js
import axios from 'axios'

const Category_BASE_API_URL = "http://localhost:8080/category";

class categoryService{
    getAll(){
        return axios.get(Category_BASE_API_URL + "/all");
    }

    addCategory(Category){
        return axios.post(Category_BASE_API_URL + "/addCategory" ,Category);
    }

    getCategoryById(id){
        return axios.get(Category_BASE_API_URL + "/getCategory/" + id);
    }

    updateCategory(id, Category){
        return axios.put(Category_BASE_API_URL + "/update/" + id,Category);
    }

    deleteCategory(id){
        return axios.delete(Category_BASE_API_URL + "/delete/" + id);
    }
    getQuizzesByCategory(id){
        return axios.get(Category_BASE_API_URL + "/" + id + "/quizzes");
    }
    
}
export default new categoryService();